src/dodal/devices/i22/fswitch.py

```diff
@@ -0,0 +1,43 @@
+import asyncio
+import time
+from enum import Enum
+from typing import Dict
+
+from bluesky.protocols import Reading
+from ophyd_async.core import StandardReadable
+from ophyd_async.core.device import DeviceVector
+from ophyd_async.epics.signal import epics_signal_r
+
+
+class FilterState(str, Enum):
+    IN = "IN"
+    OUT = "OUT"
+
+
+class FSwitch(StandardReadable):
+    """
+    Device for i22's fswitch. This currently only implements the minimum
+    functionality for retrieving the number of lenses inserted.
+
+    Eventually this should be combined with the transfocator device in the i04
+    module but is currently incompatible as the Epics interfaces are different.
+    """
+
+    NUM_FILTERS = 128
+
+    def __init__(self, prefix: str, name: str = "") -> None:
+        self.filters = DeviceVector(
+            {
+                i: epics_signal_r(FilterState, f"{prefix}FILTER-{i:03}:STATUS_RBV")
+                for i in range(FSwitch.NUM_FILTERS)
+            }
+        )
+        super().__init__(name)
+
+    async def read(self) -> Dict[str, Reading]:
+        await super().read()
+        result = await asyncio.gather(
+            *(filter.get_value() for filter in self.filters.values())
+        )
+        num_in = sum(r.value == FilterState.IN for r in result)
```

Comment:
Be worth a comment explaining that IN seems to mean "In the box", as opposed to "In the beam"

Reply:
result.count(FilterState.IN)

?

Reply:
Could also just rename the enum values to be more explicit?

Reply:
After discussion with controls, I have switched the values over.